File: sdk/src/main/java/org/coralibre/android/sdk/internal/database/persistent/DaoTEK.java
package org.coralibre.android.sdk.internal.database.persistent;


import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import org.coralibre.android.sdk.internal.database.persistent.entity.EntityTemporaryExposureKey;
import org.coralibre.android.sdk.internal.datatypes.ENInterval;

import java.util.List;

@Dao
public interface DaoTEK {

    @Insert(onConflict = OnConflictStrategy.ABORT)
    public void insertTEK(EntityTemporaryExposureKey tek);


    @Query("SELECT * FROM EntityTemporaryExposureKey")
    public List<EntityTemporaryExposureKey> getAllGeneratedTEKs();


    @Query("DELETE FROM EntityTemporaryExposureKey WHERE interval < :minKeepENIntervalNumber")
    public void truncateOldData(long minKeepENIntervalNumber);


    @Query("SELECT * FROM EntityTemporaryExposureKey WHERE interval = :interval")
    public List<EntityTemporaryExposureKey> getTekByEnNumber(ENInterval interval);


    @Query("DELETE FROM EntityTemporaryExposureKey")
    public void clearAllData();

}

File: sdk/src/main/java/org/coralibre/android/sdk/internal/database/persistent/DaoTEK.kt
package org.coralibre.android.sdk.internal.database.persistent

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import org.coralibre.android.sdk.internal.database.persistent.entity.EntityTemporaryExposureKey
import org.coralibre.android.sdk.internal.datatypes.ENInterval

@Dao
interface DaoTEK {
    @Insert(onConflict = OnConflictStrategy.ABORT)
    fun insertTEK(tek: EntityTemporaryExposureKey)

    @get:Query("SELECT * FROM EntityTemporaryExposureKey")
    val allGeneratedTEKs: List<EntityTemporaryExposureKey>

    @Query("DELETE FROM EntityTemporaryExposureKey WHERE interval < :minKeepENIntervalNumber")
    fun truncateOldData(minKeepENIntervalNumber: Long)

    @Query("SELECT * FROM EntityTemporaryExposureKey WHERE interval = :interval")
    fun getTekByEnNumber(interval: ENInterval?): List<EntityTemporaryExposureKey>

    @Query("DELETE FROM EntityTemporaryExposureKey")
    fun clearAllData()
}
